src/renderer/js/redux/actions/shape_shift.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+import Promise from "bluebird";
+import * as types from "constants/action_types";
+import { coinRegexPatterns } from "util/shape_shift";
+import type {
+  GetSupportedCoinsSuccess,
+  GetCoinStatsStart,
+  GetCoinStatsSuccess,
+  GetCoinStatsFail,
+  PrepareShapeShiftSuccess,
+  PrepareShapeShiftFail,
+  GetActiveShiftSuccess,
+  GetActiveShiftFail,
+} from "redux/reducers/shape_shift";
+import type { FormikActions } from "types/common";
+
+// use promise chains instead of callbacks for shapeshift api
+const shapeShift = Promise.promisifyAll(require("shapeshift.io"));
+
+// All ShapeShift actions
+// Action types defined in the reducer will contain some payload
+export type Action =
+  | { type: types.GET_SUPPORTED_COINS_START }
```

Comment:
I get that you are defining these here so as not to create extra boilerplate work, but I wonder if all of the types should be defined in the same place? I think this is probably best, but I just wanted to throw this question out there in case anyone has any thoughts on it

Reply:
My proposed design is:

1) Add a `types` folder inside of `js`. `types` will eventually swallow `constants`.
2) An `Action` type is defined in `Action.js` inside of this folder.
3) This file also includes all action constant definitions.
4) The `Action` type explicitly enumerates all possible action type values.

My biggest question about this design is whether there should only be one action type, or whether we should use subtypes for each possible action and payload signature.

Reply:
Possibly better to discuss this here https://github.com/lbryio/lbry-app/issues/810

Reply:
Let's not block this PR on getting action definition exactly correct.